kythe/go/platform/tools/kzip_validator/kzip_validator.go

```diff
@@ -25,24 +25,42 @@ package main
 import (
 	"flag"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"log"
+	"net/http"
+	"net/url"
 	"os"
+	"path"
 	"path/filepath"
 	"strings"
 
 	"kythe.io/kythe/go/extractors/validation"
 
 	"bitbucket.org/creachadair/stringset"
+	"github.com/mholt/archiver"
 )
 
 var (
 	kzip      = flag.String("kzip", "", "The comma separated list of kzip files to check")
 	repoURL   = flag.String("repo_url", "", "The repo to compare against")
-	version   = flag.String("version", "", "The version of the remote repo to compare")
 	localRepo = flag.String("local_repo", "", "The path of an optional local repo to specify instead of -repo_url")
 	lang      = flag.String("lang", "", "The comma-separated language file extensions to check, e.g. 'java' or 'cpp,h'")
 
 	missingFile = flag.String("missing_file", "", "An optional file to write all missing filepaths to")
+
+	// These flags use the url() function below to download the archive as:
+	// <repo_url>/<archive_prefix>/<version><archive_format>
+	//
+	// For example: -repoURL http://github.com/google/guava
+	// -> http://github.com/google/guava/archive/master.zip
+	//
+	// The supported archiveFormats are based on the dependent archiver lib:
+	// https://github.com/mholt/archiver#supported-archive-formats
+	version       = flag.String("version", "master", "The version of the remote repo to compare")
```

Comment:
Since these flags are so intertwined, it might be nice to create a struct for them:

```
type remoteRepoConfig struct {
  URL string
  ArchivePrefix string
  ...
}

var remoteRepo remoteRepoConfig

func init() {
  flag.StringVar(&remoteRepo, "remote_url", "", "The repo to compare against")
  flag.StringVar(...)
  ..."The repo to compare against"
}
```

Reply:
LGTM